packages/graphiql/src/api/hooks/useQueryParams.ts

```diff
@@ -0,0 +1,26 @@
+import { useSessionContext } from '../providers/GraphiQLSessionProvider';
+
+export function updateURL(parameters: { [name: string]: string }) {
+  const newSearch =
+    '?' +
+    Object.keys(parameters)
+      .filter(function (key) {
+        return Boolean(parameters[key]);
+      })
+      .map(function (key) {
+        return (
+          encodeURIComponent(key) + '=' + encodeURIComponent(parameters[key])
+        );
+      })
+      .join('&');
+  history.replaceState(null, null, newSearch);
+}
+
+export default function useQueryParams() {
```

Comment:
Think this should return the query string parameters, not the operational data. 
And @acao had the suggestion to also perhaps return a setter here as well.